ddtrace/contrib/grpc/server_interceptor.py

```diff
@@ -0,0 +1,32 @@
+import grpc
+
+from ddtrace import Pin
+from ...propagation.http import HTTPPropagator
+
+class GrpcServerInterceptor(grpc.ServerInterceptor):
+
+    def __init__(self):
+        self._pin = Pin.get_from(grpc)
+        self._propagator = HTTPPropagator()
+
+    def intercept_service(self, continuation, handler_call_details):
+        if not self._pin or not self._pin.enabled():
+            return continuation(handler_call_details)
+
+        self.extract_and_set_context(handler_call_details.invocation_metadata)
+
+        with self.start_span(handler_call_details.method) as span:
+            try:
+                return continuation(handler_call_details)
+            except:
+                span.set_traceback()
+                raise
+
+    def extract_and_set_context(self, metadata):
+
+        context = self._propagator.extract(dict(metadata))
```

Comment:
We probably want to have a configuration setting asking them if they want to enable distributed tracing or not (this is what we do in most other integrations right now).

Reply:
Same likely goes for the client interceptor as well?

Reply:
Do you have any example of such configuration system ?

Reply:
https://github.com/DataDog/dd-trace-py/blob/7db12f1c398c07cd5baf91c571aed672dbb6496d/ddtrace/contrib/requests/patch.py#L15-L20

https://github.com/DataDog/dd-trace-py/blob/7db12f1c398c07cd5baf91c571aed672dbb6496d/ddtrace/contrib/requests/connection.py#L63-L67